chain-abci/src/storage/account/tree.rs

```diff
@@ -186,68 +186,48 @@ pub struct TreeNode {
     pub node: NodeVariant<TreeBranch, TreeLeaf, TreeData>,
 }
 
-impl Encodable for TreeNode {
-    fn rlp_append(&self, s: &mut RlpStream) {
-        match &self.node {
-            NodeVariant::Branch(tb) => {
+impl ScaleEncode for TreeNode {
```

Comment:
Some of the variants' inner types are imported (from starling crate) and don't have that, so it can't be auto-derived. Do you think it'll be better to have all custom types instead and auto-derive all?

Reply:
I’d vote for auto-derive unless we have usecases which cannot be auto-derived such as this or where data validation is needed during deserialization.